expression/builtin_miscellaneous.go

```diff
@@ -862,6 +862,47 @@ func (c *isUsedLockFunctionClass) getFunction(ctx sessionctx.Context, args []Exp
 	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "IS_USED_LOCK")
 }
 
+type isUUIDFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *isUUIDFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETInt, types.ETString)
+	if err != nil {
+		return nil, err
+	}
+	bf.tp.Flen = 1
+	sig := &builtinIsUUIDSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_IsUUID)
+	return sig, nil
+}
+
+type builtinIsUUIDSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinIsUUIDSig) Clone() builtinFunc {
+	newSig := &builtinIsUUIDSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+// evalInt evals a builtinIsUUIDSig.
+// See https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_is-uuid
+func (b *builtinIsUUIDSig) evalInt(row chunk.Row) (int64, bool, error) {
+	val, isNull, err := b.args[0].EvalString(b.ctx, row)
+	if err != nil || isNull {
+		return 0, isNull, err
+	}
+	if _, err = uuid.Parse(val); err != nil {
```

Comment:
[`uuid.Parse`](https://pkg.go.dev/github.com/google/uuid@v1.1.2#Parse) accepts input in the form `urn:uuid:*`. So both TiDB's `IS_UUID()` and `UUID_TO_BIN()` are going to recognize this:

```sql
TiDB> select is_uuid('urn:uuid:99a9ad03-5298-11ec-8f5c-00ff90147ac3');
1
TiDB> select uuid_to_bin('urn:uuid:99a9ad03-5298-11ec-8f5c-00ff90147ac3');
0x99A9AD03529811EC8F5C00FF90147AC3
```

while MySQL will fail

```sql
TiDB> select is_uuid('urn:uuid:99a9ad03-5298-11ec-8f5c-00ff90147ac3');
0
TiDB> select uuid_to_bin('urn:uuid:99a9ad03-5298-11ec-8f5c-00ff90147ac3');
ERROR 1411 (HY000): Incorrect string value: 'urn:uuid:99a9ad03-5298-11ec-8f5c-00ff90147ac3' for function uuid_to_bin
```

should we keep this incompatibility (since the `UUID_TO_BIN()` behavior has already been part of 5.2.0)

Reply:
Also it seems that `{99a9ad03-5298-11ec-8f5c-00ff90147ac30` is wrongly considered as valid UUID due to the implementation.
```
mysql> select is_uuid('{99a9ad03-5298-11ec-8f5c-00ff90147ac30');
1
```
While in MySQL 8.0
```
mysql> select is_uuid('{99a9ad03-5298-11ec-8f5c-00ff90147ac30');
0
```

Reply:
yeah that is google/uuid#60

Reply:
Maybe you could add `{99a9ad03-5298-11ec-8f5c-00ff90147ac30` test with some comments.

Reply:
Added